provider/cloudflare/cloudflare_regional.go

```diff
@@ -72,89 +85,93 @@ func updateDataLocalizationRegionalHostnameParams(rhc regionalHostnameChange) cl
-			err := p.Client.CreateDataLocalizationRegionalHostname(ctx, resourceContainer, regionalHostnameParam)
-			if err != nil {
-				var apiErr *cloudflare.Error
-				if errors.As(err, &apiErr) && apiErr.StatusCode == http.StatusConflict {
-					log.WithFields(logFields).Debug("Regional hostname already exists, updating instead")
-					params := updateDataLocalizationRegionalHostnameParams(rhChange)
-					err := p.Client.UpdateDataLocalizationRegionalHostname(ctx, resourceContainer, params)
-					if err != nil {
-						failedChange = true
-						log.WithFields(logFields).Errorf("failed to update regional hostname: %v", err)
-					}
-					continue
-				}
-				failedChange = true
-				log.WithFields(logFields).Errorf("failed to create regional hostname: %v", err)
-			}
-		case cloudFlareUpdate:
-			log.WithFields(logFields).Debug("Updating regional hostname")
-			if p.DryRun {
-				continue
-			}
-			regionalHostnameParam := updateDataLocalizationRegionalHostnameParams(rhChange)
-			err := p.Client.UpdateDataLocalizationRegionalHostname(ctx, resourceContainer, regionalHostnameParam)
-			if err != nil {
-				var apiErr *cloudflare.Error
-				if errors.As(err, &apiErr) && apiErr.StatusCode == http.StatusNotFound {
-					log.WithFields(logFields).Debug("Regional hostname not does not exists, creating instead")
-					params := createDataLocalizationRegionalHostnameParams(rhChange)
-					err := p.Client.CreateDataLocalizationRegionalHostname(ctx, resourceContainer, params)
-					if err != nil {
-						failedChange = true
-						log.WithFields(logFields).Errorf("failed to create regional hostname: %v", err)
-					}
-					continue
-				}
-				failedChange = true
-				log.WithFields(logFields).Errorf("failed to update regional hostname: %v", err)
-			}
-		case cloudFlareDelete:
-			log.WithFields(logFields).Debug("Deleting regional hostname")
-			if p.DryRun {
-				continue
-			}
-			err := p.Client.DeleteDataLocalizationRegionalHostname(ctx, resourceContainer, rhChange.Hostname)
-			if err != nil {
-				var apiErr *cloudflare.Error
-				if errors.As(err, &apiErr) && apiErr.StatusCode == http.StatusNotFound {
-					log.WithFields(logFields).Debug("Regional hostname does not exists, nothing to do")
-					continue
-				}
-				failedChange = true
-				log.WithFields(logFields).Errorf("failed to delete regional hostname: %v", err)
-			}
+		if !p.submitRegionalHostnameChange(ctx, rhChange, resourceContainer) {
+			failedChange = true
 		}
 	}
 
 	return !failedChange
 }
 
+// submitRegionalHostnameChange applies a single regional hostname change, returns false if it fails
+func (p *CloudFlareProvider) submitRegionalHostnameChange(ctx context.Context, rhChange regionalHostnameChange, resourceContainer *cloudflare.ResourceContainer) bool {
+	logFields := log.Fields{
+		"hostname":   rhChange.Hostname,
+		"region_key": rhChange.RegionKey,
+		"action":     rhChange.action,
+		"zone":       resourceContainer.Identifier,
+	}
+	switch rhChange.action {
```

Comment:
Any chance to extract dry run and simplify logging, somethign like that

```
        logDebug := func(msg string, action) {
		log.WithFields(logFields).Debug(msg & action)
	}

       logError = .....

	if p.DryRun {
		logAction("Dry run: skipping regional hostname change", rhChange.action)
		return true
	}
```

Reply:
I found this a bit clumsy as well when writting it but couldn't find a better solution at the time.

Is the following better?
```
	log := log.WithFields(log.Fields{
		"hostname":   rhChange.Hostname,
		"region_key": rhChange.RegionKey,
		"action":     rhChange.action,
		"zone":       resourceContainer.Identifier,
	})
	if p.DryRun {
		log.Debug("Dry run: skipping regional hostname change", rhChange.action)
		return true
	}
    ...
    log.Debug("Creating regional hostname")
```
or should I avoid overriding the log package by a variable ?

Reply:
Yeap. Variable shadowing better to avoid if we could